app-modules/student-data-model/src/Filament/Resources/EducatableResource/Pages/Concerns/CanManageEducatableCareTeam.php

```diff
@@ -100,7 +100,7 @@ public function table(Table $table): Table
                             ->label('User')
                             ->searchable()
                             ->required()
-                            ->options(User::query()->tap(new HasLicense(Student::getLicenseType()))->whereDoesntHave('studentCareTeams')->pluck('name', 'id')),
+                            ->options(User::query()->tap(new HasLicense(Student::getLicenseType()))->whereDoesntHave('studentCareTeams', fn($query) => $query->where('educatable_id', $this->getOwnerRecord()->getKey()))->pluck('name', 'id')),
```

Comment:
There is so little chance of a collision here. But just to be fully safe, let's add an additional where to ensure that the type is the same as the owner record as well.